code_generator_helpers/module.py

```diff
@@ -815,16 +813,14 @@ def _emit_parsing_code(self, fields):
                         self.out("while !remaining.is_empty() {")
 
                     with Indent(self.out):
-                        self.out("let (v, new_remaining) = %s::try_parse(%s)?;", rust_type, ", ".join(try_parse_args))
+                        self.out("let (v, remaining) = %s::try_parse(%s)?;", rust_type, ", ".join(try_parse_args))
                         self.out("%s.push(v);", field_name)
-                        self.out("remaining = new_remaining;")
```

Comment:
At least this one needs to stay the way it is currently. I do not know if/where other loops in the generated code are hiding.